lib/elixir/lib/module.ex

```diff
@@ -1190,8 +1190,12 @@ defmodule Module do
     end
   end
 
-  defp preprocess_attribute(:on_load, atom) when is_atom(atom) do
-    {atom, 0}
+  defp preprocess_attribute(:on_load, value) do
```

Comment:
Got it, updated. I didn't follow your suggestion but went with a single clause where we validate the value, similar to what we do for other attributes. Let me know if you prefer your suggestion and I'll change it :)